crates/cargo-test-macro/src/lib.rs

```diff
@@ -59,6 +61,56 @@ pub fn cargo_test(attr: TokenStream, item: TokenStream) -> TokenStream {
     ret
 }
 
+#[proc_macro]
+pub fn find_tests(input: TokenStream) -> TokenStream {
+    let mut ret = TokenStream::new();
+    // remove quotations from the input string
+    let input_dir = input.to_string().replace("\"", "");
+    let input_path = Path::new(&input_dir);
+    let walker = WalkDir::new(&input_path).max_depth(3).into_iter();
+    for entry in walker.filter_entry(|e| !is_hidden(e) && is_test_or_dir(e)) {
```

Comment:
`walk_dir` is recursively searching up to the max depth of 3.
- This could pick things up at a lower depth.  That could pick up files in `add-basic.in` for example or any other of other unintended cases
- This is making a lot of assumptions that aren't clear to the caller.

Reply:
I agree that it makes a lot of assumptions about the caller, how should I go about documenting it? As for the max depth of 3, I don't think it goes past `tests/snapshots/{test file}/{test name}/test.rs`. It looks like `tests/snapshots/` is 1, `tests/snapshots/{test file}/` is 2 and `tests/snapshots/{test file}/{test name}/`is 3. That should make it so it can't find anything below `test.rs` and it forces `test.rs` to be where it is

Reply:
One of the concerns I brought up is with having files at a lower depth.

If we move the macro call to `cargo_add.rs`, causing the macro to get a more specific path and rename `test.rs` to `mod.rs`, I think we could not use a third party library at all.  We can just list the directory given and if any directories in it have a `mod.rs`, we then provide that.

Reply:
I'll change it to that then